sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -180,4 +184,52 @@ export class ApplicationController extends BaseController {
       );
     }
   }
+
+  /**
+   * Generic method to update all Student Application status from frontend.
+   * @param applicationId application id to be updated.
+   * @body payload contains the status, that need to be updated
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/update-status")
+  async updateStudentApplicationStatus(
+    @UserToken() userToken: IUserToken,
+    @Param("applicationId") applicationId: number,
+    @Body() payload: ApplicationStatusToBeUpdatedDto,
+  ): Promise<void> {
+    const studentApplication =
+      await this.applicationService.getApplicationByIdAndUserName(
+        applicationId,
+        userToken.userName,
+      );
+
+    if (!studentApplication) {
+      throw new UnprocessableEntityException(
+        `Application ${applicationId} associated with requested student does not exist.`,
+      );
+    }
+    // delete workflow if the payload status is cancelled
+    // workflow doest not exists for draft application
+    if (
+      payload.applicationStatus === ApplicationStatus.cancelled &&
```

Comment:
Just a suggestion, always try to have the constant first, in a IF check (constant === variable). 

if(payload && ApplicationStatus.cancelled === payload.applicationStatus && ApplicationStatus.draft !== studentApplication.applicationStatus && studentApplication.assessmentWorkflowId) {

}

Reply:
mm, Yoda notation, it's just a style of programming to avoid errors, ya sure, we can follow that

Reply:
okay, I thought that since keys in the interface are mandatory, it will fail if a null payload comes. okay, it's not true, will add the condition for that